scripts/generate_errors.pl

```diff
@@ -108,12 +148,23 @@
 
     if ($found_ll)
     {
+        if ($error_number < 1 || $error_number > 0x7f)
+        {
+            printf STDERR ("Error: invalid low-level error code $error_code ($error_name)\n");
+            exit 1;
+        }
         $code_check = \$ll_code_check;
         $old_define = \$ll_old_define;
         $white_space = '    ';
     }
     else
     {
+        if ($error_number == 0 || $error_number > 0x7fff ||
```

Comment:
Do we want to also catch when an error number doesn't have the same high level module ID as other errors within the same module (that have been seen so far)?

Note that we may need to make an exception for errors in error number allocation we've already made (like `MBEDTLS_ERR_X509_FATAL_ERROR` which doesn't share the same high level module ID as other X.509 errors).